### PR TITLE
Housing dashboard style

### PR DESCRIPTION
**Pull request by cbourgois**
## Update Housing dashboard to use ui-kit

Update Housing dashboard to use ui-kit tiles (based on ui-kit-bs tiles)

## Related PRs:

* #266 should be merged (this branch is based on)
* ovh-ux/ovh-ui-kit-bs#52 should be merged
* Some changes will be needed when #214 will be merged

### CODE REVIEW

File: client/app/css/less/common/tile.less [hunk 1]
```diff
@@ -0,0 +1,5 @@
+.oui-tile {
```
Comment (antleblanc) [inline client/app/css/less/common/tile.less]: Should be placed on `ovh-ux/ovh-ui-kit-bs` instead?

File: client/app/components/expiration/service-expiration-date.component.html [hunk 2]
```diff
@@ -1,17 +1,32 @@
...
+</a>
+<oui-action-menu
+    data-ng-if="$ctrl.useUiKit && $ctrl.constants.target !== 'US' && !$ctrl.forceHideRenewAction && ((!$ctrl.isAutoRenew() && $ctrl.getRenewUrl() && !$ctrl.hideRenewAction) || ($ctrl.serviceInfos.status === 'expired'))"
+    aria-label="{{:: tr('menu_action_label', tr('core_service_expiration_label')) }}"
+    align="end" compact>
+    <oui-action-menu-item
+        data-text="{{:: tr('core_service_renew_button') }}"
+        data-aria-label="{{:: tr('core_service_renew_button') }}"
+        data-href="{{:: $ctrl.getRenewUrl() }}"></oui-action-menu-item>
+</oui-action-menu>
```
Comment (antleblanc) [inline client/app/components/expiration/service-expiration-date.component.html]: `insert_final_newline = true` :)

File: client/app/components/expiration/service-expiration-date.component.html [hunk 3]
```diff
@@ -1,17 +1,32 @@
...
+<a
+    class="oui-link"
+    data-ng-href="{{:: $ctrl.getRenewUrl() }}"
+    data-ng-bind=":: $ctrl.serviceInfos.expiration | date:'mediumDate'"
+    data-ng-if="$ctrl.useUiKit && $ctrl.constants.target !== 'US' && !$ctrl.hideRenewDate">
+</a>
+<oui-action-menu
+    data-ng-if="$ctrl.useUiKit && $ctrl.constants.target !== 'US' && !$ctrl.forceHideRenewAction && ((!$ctrl.isAutoRenew() && $ctrl.getRenewUrl() && !$ctrl.hideRenewAction) || ($ctrl.serviceInfos.status === 'expired'))"
+    aria-label="{{:: tr('menu_action_label', tr('core_service_expiration_label')) }}"
+    align="end" compact>
```
Comment (antleblanc) [inline client/app/components/expiration/service-expiration-date.component.html]: `data-align` & `data-compact`.

File: client/app/components/expiration/service-expiration-date.component.html [hunk 4]
```diff
@@ -1,17 +1,32 @@
-<span data-ng-class="{ 'd-block': !$ctrl.inline }">
-    <span data-ng-if="$ctrl.constants.target !== 'US'">
-        <a class="mr-2"
-            data-ng-href="{{:: $ctrl.getRenewUrl() }}"
-            data-ng-bind=":: $ctrl.serviceInfos.expiration | date:'mediumDate'"
-            data-ng-if="!$ctrl.hideRenewDate">
-        </a>
+<span data-ng-if="!$ctrl.useUiKit && $ctrl.constants.target !== 'US'"
+    data-ng-class="{ 'd-block': !$ctrl.inline }">
```
Comment (antleblanc) [inline client/app/components/expiration/service-expiration-date.component.html]: Attributes are not well aligned now :/

File: client/app/dedicated/server/statistics/usage.html [hunk 5]
```diff
@@ -1,33 +1,35 @@
...
+                            class="control-label"
+                            data-i18n-static="server_tab_stats_rtm_label_info">
                         </label>
                         <select class="form-control"
-                                id="info"
-                                name="selectRtmOption"
-                                data-ng-options="tr('server_tab_stats_rtm_options_label_' + option.label) for option in optionsRtm"
-                                data-ng-model="rtmOptions.value"
-                                data-ng-change="changeRtmOptions()">
+                            id="info"
```
Comment (antleblanc) [inline client/app/dedicated/server/statistics/usage.html]: Attributes are not well aligned now :/

File: client/app/dedicated/server/statistics/infos.html [hunk 6]
```diff
@@ -1,108 +1,105 @@
...
-                                    <span data-i18n-static="server_tab_stats_rtm_infos_cpu_core"></span> :
-                                    <span data-ng-bind="infoRtm.cpu.core || '-'"></span><br />
-                                    <span data-i18n-static="server_tab_stats_rtm_infos_cpu_cache"></span> :
-                                    <span data-ng-bind="infoRtm.cpu.cache.value + infoRtm.cpu.cache.unit"></span>
+                    <div class="oui-tile__item">
+                        <dl class="oui-tile__definition">
+                            <dt class="oui-tile__term"
+                                data-i18n-static="server_tab_stats_rtm_infos_cpu"></dt>
+                            <dd class="oui-tile__description">
+                                <span data-ng-bind="infoRtm.cpu.name || '-'"></span><br />
```
Comment (antleblanc) [inline client/app/dedicated/server/statistics/infos.html]: You can apply `d-block` on the `<span>` element and get rid of `<br/>`.

File: client/app/dedicated/server/location/dedicated-server-location.html [hunk 7]
```diff
@@ -1,30 +1,32 @@
...
+</div>
+<div class="oui-tile__item">
+    <dl class="oui-tile__definition">
+        <dt class="oui-tile__term"
+            data-i18n-static="server_configuration_server_id"></dt>
+        <dd class="oui-tile__description"
+            data-ng-bind="infoServer.serverId || '-'"></dd>
+    </dl>
+</div>
+
```
Comment (antleblanc) [inline client/app/dedicated/server/location/dedicated-server-location.html]: `insert_final_newline = true` (just one)

File: client/app/dedicated/server/dashboard/DASHBOARD_VIEW.html [hunk 8]
```diff
@@ -2,573 +2,607 @@
...
+                                <span class="oui-status oui-status_warning"
+                                    data-ng-if="!server.os" data-i18n-static="server_configuration_distribution_none">
+                                </span>
+                            </dd>
+                        </dl>
+                        <div class="oui-tile__actions"
+                            data-ng-if="!server.isExpired && server.commercialRange !== 'housing'">
+                            <oui-action-menu
+                                aria-label="{{:: tr('menu_action_label', tr('server_configuration_distribution')) }}"
+                                align="end" compact>
```
Comment (antleblanc) [inline client/app/dedicated/server/dashboard/DASHBOARD_VIEW.html]: `data-align` & `data-compact`.

File: client/app/dedicated/server/dashboard/DASHBOARD_VIEW.html [hunk 9]
```diff
@@ -2,573 +2,607 @@
...
+                                        'oui-status_error': server.state == 'HACKED' || server.state == 'HACKED_BLOCKED'
+                                    }" data-ng-bind=":: tr('server_configuration_state_' + server.state)">
+                                </span>
+                            </dd>
+                        </dl>
+                        <div class="oui-tile__actions"
+                            data-ng-if="server.state == 'HACKED' || server.state == 'HACKED_BLOCKED'">
+                            <oui-action-menu
+                                aria-label="{{:: tr('menu_action_label', tr('server_configuration_state_title')) }}"
+                                align="end" compact>
```
Comment (antleblanc) [inline client/app/dedicated/server/dashboard/DASHBOARD_VIEW.html]: `data-align` & `data-compact`.

File: client/app/dedicated/server/consumption/consumption.html [hunk 10]
```diff
@@ -1,68 +1,76 @@
...
+                            </div>
+                        </div>
+                        <span data-ng-bind="tr('server_consumption_current_period_remaining', [ctrl.traffic.data.usage.remainingOutputQuotaSize.text])"></span>
+                    </dd>
+                </dl>
+                <div class="oui-tile__actions"
+                    data-ng-if="ctrl.canOrderTraffic() && ctrl.canOrderMoreTraffic()">
+                    <oui-action-menu
+                        aria-label="{{:: tr('menu_action_label', tr('server_consumption_current_period_label')) }}"
+                        align="end" compact>
```
Comment (antleblanc) [inline client/app/dedicated/server/consumption/consumption.html]: `data-align` & `data-compact`.

File: client/app/dedicated/server/consumption/consumption.html [hunk 11]
```diff
@@ -1,68 +1,76 @@
...
-                 aria-valuenow="{{ ctrl.traffic.data.usage.outputQuotaUsed.value }}"
-                 aria-valuemin="0"
-                 aria-valuemax="{{ ctrl.traffic.data.usage.outputQuotaSize.value }}"
-                 data-ng-style="{ 'width': ctrl.traffic.data.usage.remainingOutputQuotaSize.percent + '%' }">
-            </div>
+    <div class="oui-tile">
+        <h4 class="oui-header_4 oui-tile__title">
+            <span data-i18n-static="server_consumption_label"></span>
+            <i class="oui-icon oui-icon-help_circle oui-tooltip__trigger"
+                oui-tooltip="{{:: tr('server_consumption_explanation') }}"></i>
```
Comment (antleblanc) [inline client/app/dedicated/server/consumption/consumption.html]: +`data-` and prefer using a `<span>` tag instead?

**Comment by cbourgois** [comment]
Done ;)

File: client/app/dedicated/housing/dashboard/dedicated-housing-dashboard.html [hunk 12]
```diff
@@ -3,161 +3,176 @@
...
-                            <dt data-i18n-static="housing_creation_date"></dt>
-                            <dd data-ng-bind="housing.creation | date:'mediumDate'"></dd>
-                        </div>
-                    </li>
-                </ul>
+                        </dd>
+                    </dl>
+                </div>
+                <div class="oui-tile__item">
+                    <dl class="oui-tile__definition">
```
Comment (antleblanc) [inline client/app/dedicated/housing/dashboard/dedicated-housing-dashboard.html]: I don't know if it's related to the style you want to apply but in fact you are creating some `<dl>` lists that always have a single item in it.

**Pull request merged**